panel/widgets/file_selector.py

```diff
@@ -100,7 +101,7 @@ class FileSelector(CompositeWidget):
 
     _composite_type = Column
 
-    def __init__(self, directory=None, **params):
+    def __init__(self, directory: Union[AnyStr, os.PathLike[AnyStr]]=None, **params):
```

Comment:
I put in the type I believe the will work from directory from looking at the code. But I am in doubt if this is the type hint your want @philippjfr